### PR TITLE
обновлены настройки в settings.json

### PR DESCRIPTION
**Pull request by itsaisina**

### CODE REVIEW

**Comment by demid5111** [comment]
nothing to review to this moment. btw, write commit messages in English

**Comment by demid5111** [comment]
nothing to review to this moment

**Comment by demid5111** [comment]
nothing to review to this moment

**Comment by itsaisina** [comment]
sorry, I don't know why, but I have no opportunity to change the label to "review required" 😿  maybe I'm too silly to properly interact with github

File: lab_7_llm/assets/main.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        try {
+            const response = await fetch("/infer", {
+                method: "POST",
+                headers: { "Content-Type": "application/json" },
+                body: JSON.stringify({ question: query })
+            });
+
+            const result = await response.json();
+            outputField.textContent = "AI says: " + result.infer;
```
Comment (demid5111) [inline lab_7_llm/assets/main.js:28]: use string interpolation

File: lab_7_llm/assets/styles.css [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    font-size: 16px;
+    outline: none;
+    transition: border-color 0.3s ease-in-out;
+}
+
+input:focus {
+    border-color: #ff80bf;
+}
+
+button {
```
Comment (demid5111) [inline lab_7_llm/assets/styles.css:35]: all styles should be applied by class, not element type

File: lab_7_llm/main.py [hunk 3]
```diff
@@ -35,12 +53,32 @@ def analyze(self) -> dict:
         Returns:
             dict: Dataset key properties
         """
+        dataset_properties = {
```
Comment (demid5111) [inline lab_7_llm/main.py:56]: return immediately

File: lab_7_llm/main.py [hunk 4]
```diff
@@ -35,12 +53,32 @@ def analyze(self) -> dict:
...
 
     @report_time
     def transform(self) -> None:
         """
         Apply preprocessing transformations to the raw dataset.
         """
+        drop_cols = ['id', 'severe_toxic', 'obscene', 'threat', 'insult', 'identity_hate']
+        self._raw_data.drop(columns=drop_cols, errors='ignore', inplace=True)
+
+        self._raw_data.rename(columns={
```
Comment (demid5111) [inline lab_7_llm/main.py:75]: bad that you change raw data - by that mutating original data that you might want to use in future

File: lab_7_llm/main.py [hunk 5]
```diff
@@ -74,6 +114,8 @@ def __getitem__(self, index: int) -> tuple[str, ...]:
         Returns:
             tuple[str, ...]: The item to be received
         """
+        row = self._data.iloc[index]
+        return row[ColumnNames.SOURCE.value], str(row[ColumnNames.TARGET.value])
```
Comment (demid5111) [inline lab_7_llm/main.py:118]: why do you need target?

File: lab_7_llm/main.py [hunk 6]
```diff
@@ -111,6 +158,27 @@ def analyze_model(self) -> dict:
...
             dict: Properties of a model
         """
+        dummy_inputs = torch.ones((1,
+                                  self._model.config.max_position_embeddings),
+                                  dtype=torch.long)
+
+        input_data = {'input_ids': dummy_inputs,
+                      'attention_mask': dummy_inputs}
+
+        if isinstance(self._model, torch.nn.Module):
```
Comment (demid5111) [inline lab_7_llm/main.py:168]: why do you need this condition? should be possible without it

File: lab_7_llm/main.py [hunk 7]
```diff
@@ -111,6 +158,27 @@ def analyze_model(self) -> dict:
...
+                'embedding_size': self._model.config.max_position_embeddings,
+                'output_shape': model_summary.summary_list[-1].output_size,
+                'num_trainable_params': model_summary.trainable_params,
+                'vocab_size': self._model.config.vocab_size,
+                'size': model_summary.total_param_bytes,
+                'max_context_length': self._model.config.max_length
+            }
+            return model_properties
+
+        raise TypeError("model is not a valid torch.nn.Module")
```
Comment (demid5111) [inline lab_7_llm/main.py:181]: ?

File: lab_7_llm/main.py [hunk 8]
```diff
@@ -144,6 +226,24 @@ def _infer_batch(self, sample_batch: Sequence[tuple[str, ...]]) -> list[str]:
         Returns:
             list[str]: Model predictions as strings
         """
+        if self._model is None:
+            raise ValueError("model has not been initialized")
+
+        texts = [sample[0] for sample in sample_batch]
```
Comment (demid5111) [inline lab_7_llm/main.py:232]: exactly to my question about target column

File: lab_7_llm/main.py [hunk 9]
```diff
@@ -144,6 +226,24 @@ def _infer_batch(self, sample_batch: Sequence[tuple[str, ...]]) -> list[str]:
...
+            texts,
+            return_tensors="pt",
+            truncation=True,
+            padding=True,
+            max_length=self._max_length
+        )
+        inputs = {k: v.to(self._device) for k, v in inputs.items()}
+
+        with torch.no_grad():
+            outputs = self._model(**inputs)
```
Comment (demid5111) [inline lab_7_llm/main.py:243]: where do you put model in eval mode?

File: requirements.txt [hunk 10]
```diff
@@ -0,0 +1,10 @@
+datasets==3.2.0
+evaluate==0.4.3
+fastapi==0.115.7
+pandas==2.2.3
+pydantic==2.10.6
+scikit-learn==1.6.1
+torch==2.1.2
+torchinfo==1.8.0
+transformers==4.48.0
+uvicorn==0.34.0
```
Comment (demid5111) [inline requirements.txt:10]: newline

File: lab_7_llm/start.py [hunk 11]
```diff
@@ -2,15 +2,53 @@
...
+
+    preprocessor.transform()
+    task_dataset = TaskDataset(preprocessor.data.head(100))
+
+    pipeline = LLMPipeline(settings.parameters.model,
+                           task_dataset,
+                           max_length=120,
+                           batch_size=64,
+                           device="cpu")
+    pipeline.analyze_model()
```
Comment (demid5111) [inline lab_7_llm/start.py:40]: task for 6 implies demonstrating infer_sample as well

**Review by demid5111 — CHANGES_REQUESTED** [review]
amazing progress!

**Comment by demid5111** [comment]
about labels. is it still an issue? please write to chat with the question

File: lab_7_llm/assets/main.js (hunk 1 above)
Comment (itsaisina) [reply lab_7_llm/assets/main.js]: done

File: lab_7_llm/assets/styles.css (hunk 2 above)
Comment (itsaisina) [reply lab_7_llm/assets/styles.css]: done

File: lab_7_llm/main.py (hunk 3 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: done

File: lab_7_llm/main.py (hunk 4 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: done

File: requirements.txt (hunk 10 above)
Comment (itsaisina) [reply requirements.txt]: done

File: lab_7_llm/start.py (hunk 11 above)
Comment (itsaisina) [reply lab_7_llm/start.py:40]: done

File: lab_7_llm/main.py (hunk 9 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: done

File: lab_7_llm/main.py (hunk 8 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: done

File: lab_7_llm/main.py (hunk 5 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: no, done

File: lab_7_llm/main.py (hunk 6 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: I can't pass mypy checks without this

File: lab_7_llm/main.py (hunk 7 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: I can't pass mypy checks without this

File: requirements.txt [hunk 12]
```diff
@@ -0,0 +1,10 @@
+datasets==3.2.0
+evaluate==0.4.3
+fastapi==0.115.7
+pandas==2.2.3
+pydantic==2.10.6
+scikit-learn==1.6.1
+torch==2.1.2
```
Comment (demid5111) [inline requirements.txt:7]: use cpu version of torch

File: lab_7_llm/service.py [hunk 13]
```diff
@@ -14,6 +27,46 @@ def init_application() -> tuple[FastAPI, LLMPipeline]:
...
     """
+    model_app = FastAPI()
+
+    settings = LabSettings(BASE_DIR / "settings.json")
+    dataset = TaskDataset(pd.DataFrame())
+    model_pipeline = LLMPipeline(
+        model_name=settings.parameters.model,
+        dataset=dataset,
+        max_length=120,
+        batch_size=64,
```
Comment (demid5111) [inline lab_7_llm/service.py:38]: ?

File: lab_7_llm/main.py [hunk 14]
```diff
@@ -168,3 +260,18 @@ def run(self) -> dict | None:
...
             dict | None: A dictionary containing information about the calculated metric
         """
+        data_frame = pd.read_csv(self._data_path)
+
+        targets = data_frame[ColumnNames.TARGET.value]
+        predictions = data_frame[ColumnNames.PREDICTION.value]
+
+        results = {}
+        for metric_item in self._metrics:
+            metric = load(metric_item.value)
```
Comment (demid5111) [inline lab_7_llm/main.py:270]: load metrics on init phase

File: lab_7_llm/main.py [hunk 15]
```diff
@@ -144,6 +221,19 @@ def _infer_batch(self, sample_batch: Sequence[tuple[str, ...]]) -> list[str]:
...
+
+        inputs = {k: v.to(self._device) for k, v in self._tokenizer(
+            list(sample_batch[0]),
+            return_tensors="pt",
+            truncation=True,
+            padding="max_length",
+            max_length=self._max_length
+        ).items()}
+
+        preds = torch.argmax(self._model(**inputs).logits, dim=1)
```
Comment (demid5111) [inline lab_7_llm/main.py:235]: forgot to disable gradients

File: lab_7_llm/main.py [hunk 16]
```diff
@@ -144,6 +221,19 @@ def _infer_batch(self, sample_batch: Sequence[tuple[str, ...]]) -> list[str]:
         Returns:
             list[str]: Model predictions as strings
         """
+        if self._model is None:
+            raise ValueError("model has not been initialized")
+
+        inputs = {k: v.to(self._device) for k, v in self._tokenizer(
```
Comment (demid5111) [inline lab_7_llm/main.py:227]: 1. call tokenizer
2. tokenizer result to device (as a whole will also work)

File: lab_7_llm/main.py [hunk 17]
```diff
@@ -111,6 +158,28 @@ def analyze_model(self) -> dict:
         Returns:
             dict: Properties of a model
         """
+        config = self._model.config
+        dummy_inputs = torch.ones((1,
+                                  config.max_position_embeddings),
+                                  dtype=torch.long)
+
+        input_data = {'input_ids': dummy_inputs,
```
Comment (demid5111) [inline lab_7_llm/main.py:166]: 🎖️

File: lab_7_llm/main.py [hunk 18]
```diff
@@ -35,12 +53,32 @@ def analyze(self) -> dict:
...
 
     @report_time
     def transform(self) -> None:
         """
         Apply preprocessing transformations to the raw dataset.
         """
+        transformed_data = self._raw_data.copy()
+
+        drop_cols = ['id', 'severe_toxic', 'obscene', 'threat', 'insult', 'identity_hate']
+        transformed_data.drop(columns=drop_cols, errors='ignore', inplace=True)
```
Comment (demid5111) [inline lab_7_llm/main.py:73]: true pandas code is pipelining:

```py

(
  df
    .drop(..)
    .rename(...)
)
```

**Review by demid5111 — CHANGES_REQUESTED** [review]
looks quite almost done, polish

File: lab_7_llm/main.py (hunk 6 above)
Comment (demid5111) [reply lab_7_llm/main.py]: have you tried?

File: lab_7_llm/main.py (hunk 7 above)
Comment (demid5111) [reply lab_7_llm/main.py]: ?

File: lab_7_llm/main.py (hunk 15 above)
Comment (itsaisina) [reply lab_7_llm/main.py:235]: but method is already decorated with @torch.no_grad(), so gradient computation is disabled without needing an explicit with torch.no_grad(): block, am I wrong?

File: lab_7_llm/main.py (hunk 18 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: changed

File: lab_7_llm/main.py (hunk 16 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: changed

File: lab_7_llm/main.py (hunk 14 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: changed

File: lab_7_llm/service.py (hunk 13 above)
Comment (itsaisina) [reply lab_7_llm/service.py]: changed to 1

File: requirements.txt (hunk 12 above)
Comment (itsaisina) [reply requirements.txt:7]: added

File: lab_7_llm/main.py (hunk 6 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: of course I tried, but there is: 

> "lab_7_llm/main.py:171: error: Argument 1 to "summary" has incompatible type "HF Model Like | None"; expected Module  [arg-type]"

File: lab_7_llm/main.py (hunk 7 above)
Comment (itsaisina) [reply lab_7_llm/main.py]: answered in the previous conversation

File: lab_7_llm/main.py [hunk 19]
```diff
@@ -35,12 +53,34 @@ def analyze(self) -> dict:
         Returns:
             dict: Dataset key properties
         """
+        return {
+            "dataset_number_of_samples": self._raw_data.shape[0],
```
[the remark this answers is not in the document]
Comment (itsaisina) [reply lab_7_llm/main.py]: changed

File: lab_7_llm/main.py [hunk 20]
```diff
@@ -35,12 +53,34 @@ def analyze(self) -> dict:
...
+            "dataset_columns": self._raw_data.shape[1],
```
[the remark this answers is not in the document]
Comment (itsaisina) [reply lab_7_llm/main.py]: changed

File: requirements.txt [hunk 21]
```diff
@@ -0,0 +1,12 @@
...
+torchinfo==1.8.0
+transformers==4.48.0
+uvicorn==0.34.0
+
+--extra-index-url https://download.pytorch.org/whl/cpu
```
[the remark this answers is not in the document]
Comment (itsaisina) [reply requirements.txt]: added
